lib/asciidoctor-epub3/packager.rb

```diff
@@ -459,7 +460,7 @@ def distill_epub_to_mobi epub_file, target
       require 'kindlegen' unless defined? ::Kindlegen
       kindlegen_cmd = ::Kindlegen.command
     end
-    mobi_file = ::File.basename(target).sub EpubExtensionRx, '.mobi'
+    mobi_file = ::File.basename(target =~ MobiExtensionRx ? target : %(#{::Asciidoctor::Helpers.rootname target}.mobi))
```

Comment:
Now we only substitute if the extension isn't `.mobi`. That's the one thing we are enforcing. The extension of the mobi file must be `.mobi` (case insensitive).